spec/custom_rules/AmplifyBranchBasicAuthConfigPasswordRule_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'spec_helper'
+require 'password_rule_spec_helper'
+require 'cfn-model'
+
+resource_type = 'AWS::Amplify::Branch'
+password_property = 'BasicAuthConfig'
+sub_property_name = 'Password'
+test_template_type = 'yaml'
+
+require "cfn-nag/custom_rules/#{rule_name(resource_type, password_property, sub_property_name)}"
+
+describe Object.const_get(rule_name(resource_type, password_property, sub_property_name)), :rule do
```

Comment:
this is slick but.... obscures the readability of the test.  how does this end up looking in the test report?

Reply:
Here is the output for a failing spec test. You can see it dynamically creates the context with the proper rule class name followed by the pertinent resource and property items with the description.

```
Failures:

  1) AmplifyBranchBasicAuthConfigPasswordRule AWS::Amplify::Branch BasicAuthConfig Password as a literal in plaintext returns offending logical resource id
     Failure/Error:
       expect(actual_logical_resource_ids).to eq \
         expected_logical_resource_ids(desired_test_result, resource_type)
     
       expected: ["AmplifyBranch"]
            got: []
     
       (compared using ==)
     # ./spec/password_rule_spec_helper.rb:148:in `run_test'
     # ./spec/custom_rules/AmplifyBranchBasicAuthConfigPasswordRule_spec.rb:17:in `block (4 levels) in <top (required)>'

Finished in 1.36 seconds (files took 0.91292 seconds to load)
427 examples, 1 failure

Failed examples:

rspec './spec/custom_rules/AmplifyBranchBasicAuthConfigPasswordRule_spec.rb[1:5:1]' # AmplifyBranchBasicAuthConfigPasswordRule AWS::Amplify::Branch BasicAuthConfig Password as a literal in plaintext returns offending logical resource id

Coverage report generated for RSpec to /Users/peter.helewski/Stelligent/cfn_nag/coverage. 2067 / 2139 LOC (96.63%) covered.
```